tests/IonKey.spec.ts

```diff
@@ -77,4 +77,104 @@ describe('IonKey', async () => {
       );
     });
   });
+
+  describe('isJwkEs256k()', async () => {
+    it('should return true for a JwkEs256K key', async () => {
+      const [publicKey, privateKey] = await IonKey.generateEs256kOperationKeyPair();
+      expect(IonKey.isJwkEs256k(publicKey)).toBeTruthy();
+      expect(IonKey.isJwkEs256k(privateKey)).toBeTruthy();
+    });
+
+    it('should return false for a JwkEd25519 key', async () => {
+      const [publicKey, privateKey] = await IonKey.generateEd25519OperationKeyPair();
+      expect(IonKey.isJwkEs256k(publicKey)).toBeFalsy();
+      expect(IonKey.isJwkEs256k(privateKey)).toBeFalsy();
+    });
+  });
+
+  describe('generateEd25519OperationKeyPair()', async () => {
+    it('should create a key pair successfully.', async () => {
+      const [publicKey, privateKey] = await IonKey.generateEd25519OperationKeyPair();
+
+      expect(Object.keys(publicKey).length).toEqual(3);
+      expect(Object.keys(privateKey).length).toEqual(4);
+
+      expect(publicKey.d).toBeUndefined();
+      expect(privateKey.d).toBeDefined();
+      expect(publicKey.crv).toEqual(privateKey.crv);
+      expect(publicKey.kty).toEqual(privateKey.kty);
+      expect(publicKey.x).toEqual(privateKey.x);
+    });
+  });
+
+  describe('generateEd25519DidDocumentKeyPair()', async () => {
+    it('should create a key pair successfully.', async () => {
+      const keyId = 'anyId';
+      const [didDocumentPublicKey, privateKey] = await IonKey.generateEd25519DidDocumentKeyPair({ id: keyId, purposes: [IonPublicKeyPurpose.Authentication] });
+
+      expect(didDocumentPublicKey.id).toEqual(keyId);
+      expect(didDocumentPublicKey.purposes).toEqual([IonPublicKeyPurpose.Authentication]);
+      expect(didDocumentPublicKey.type).toEqual('Ed25519VerificationKey2018');
+
+      expect(Object.keys(didDocumentPublicKey.publicKeyJwk).length).toEqual(3);
+      expect(Object.keys(privateKey).length).toEqual(4);
+
+      expect(privateKey.d).toBeDefined();
+
+      const publicKey = didDocumentPublicKey.publicKeyJwk as JwkEs256k;
```

Comment:
Why is this being typecast here? I'm thinking this workaround should probably take into account that this additional abstraction should be extended into the type system as well (e.g. change to generic Jwk type instead)

Reply:
Added more specific type for publicKeyJwk: https://github.com/decentralized-identity/ion-sdk/pull/20/commits/045c4254604d808fe0bc8e0cb8d0a3f2f96c8af6